Add implementation for fixExpectedCommaError (#25201)

adeniszczyc opened the pull request:
Code fix for changing `;` to a `,` within a list. 

Typescript issue: https://github.com/microsoft/TypeScript/issues/25201

![quick_fix](https://user-images.githubusercontent.com/2099219/65708140-d0ced980-e085-11e9-98db-0fd15d4b1808.gif)

**Describe your changes**
Adds the quick fix for changing a `;` to `,` between list items in either an object or array context by using the `_0_expected` diagnostic.

**Testing performed**
Tested in VSCode to ensure that the error generates the expected quick fix for changing a `;` to a `,` within the above context.

**Additional context**
No additional context.

joeywatts commented on src/services/codefixes/fixExpectedCommaError.ts:
```diff
@@ -0,0 +1,47 @@
+                                 isArrayLiteralExpression(node.parent);
+
+        return (node.kind === SyntaxKind.SemicolonToken &&
+                node.parent &&
```
Can you move this check up to where you declare `isCorrectContext` so that it checks that `node.parent` is defined before it checks the type of node?

adeniszczyc commented:
Thanks - I've added two test cases and moved the check for the context. Just a couple of other points:
- Should I also add any negative test case when the code fix is not applied?
- Do you think the naming `fixExpectedCommaError` is fine, especially the `Error` part of this?

joeywatts commented:
Good point - it does seem like `fixExpectedComma` might be more consistent with the existing codefix files. If that's easy enough to change, then maybe that's worth doing now before we get feedback from Microsoft (if not, we can wait for them to comment on it).

It would be good to have a test case where the `Expected ','` error shows up but the code fix is not available. For example, `{ one: 1 one }`. You can use `verify.not.codeFixAvailable("fixExpectedComma")` to do that check.

Once that's done, could you squash the commits down to 1 and sign off (that will fix that DCO thing).

```
git rebase -i HEAD~N --signoff # N is the number of commits
# opens your editor, change the "pick" on all lines after the first line to "squash", save and quit
# editor will open to allow you to change the commit message
git push --force-with-lease
```
